SportsHub/SportsHub.Api/Exceptions/ExceptionHandler.cs

```diff
@@ -10,11 +11,34 @@ public async Task InvokeAsync(HttpContext context, RequestDelegate next)
         {
             await next(context);
         }
-
-        catch (Exception e)
+        catch (BusinessLogicException ex)
+        {
+            await HandleBusinessLogicExceptionAsync(context, ex);
+        }
+        catch (NotFoundException ex)
+        {
+            await HandleNotFoundExceptionAsync(context, ex);
+        }
+        catch (Exception ex)
         {
             context.Response.StatusCode = (int)HttpStatusCode.InternalServerError;
-            await context.Response.WriteAsync(e.Message);
+            await context.Response.WriteAsync(ex.Message);
         }
     }
+
+    private async Task HandleNotFoundExceptionAsync(HttpContext context, NotFoundException ex)
+    {
+        context.Response.ContentType = "/application/json";
+        context.Response.StatusCode = ex.ErrorCode;
```

Comment:
in the Response.StatusCode = should be HTTPStatus code, 400,404, 500
Error code should be returned with the message in response bode

Reply:
Oh okey i understand and i will change it .